app/src/main/java/com/moonpig/mvisample/productdetail/ProductDetailScreenViewState.kt

```diff
@@ -0,0 +1,19 @@
+package com.moonpig.mvisample.productdetail
+
+import com.moonpig.mvisample.mvibase.BaseViewState
+
+data class ProductDetailScreenViewState(val getProductDetailInFlight: Boolean = false,
+                                        val getProductDetailError: Throwable? = null,
+                                        val productDetail: ProductDetailViewState = ProductDetailViewState.None,
+                                        val addToBasketInFlight: Boolean = false,
+                                        val addToBasketError: Throwable? = null) :
+        BaseViewState
+
+data class ProductDetailViewState(val name: String = "",
+                                  val description: String = "",
+                                  val price: Int = 0,
+                                  val imageUrl: String = ""){
+    companion object {
+        val None = ProductDetailViewState()
```

Comment:
Not too keen on this as the None state has fields that aren't meaningful. A sealed class with Product and None as children enforces the contract that this can't be displayed.

Reply:
You can alternatively make it nullable and use a `?.` call when rendering it, given `None` has no state at all.